File: Crio/crio_programming_interview_problems-master/LongestPalindromicSubstringLength/LongestPalindromicSubstringLength.py

def longestPalindromicSubstringLength(n, s):
# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    maxLength = 1
    start,low , high = 0,0,0
    for i in range(1, n): 
        low = i - 1
        high = i 
        while low >= 0 and high < n and s[low] == s[high]: 
            if high - low + 1 > maxLength: 
                start = low 
                maxLength = high - low + 1
            low -= 1
            high += 1
        low = i - 1
        high = i + 1
        while low >= 0 and high < n and s[low] == s[high]: 
            if high - low + 1 > maxLength: 
                start = low 
                maxLength = high - low + 1
            low -= 1
            high += 1
    return maxLength 
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS


def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input().rstrip()
        ans = longestPalindromicSubstringLength(n , s)
        print(ans)


if __name__ == "__main__":
    main()

File: Crio/crio_programming_interview_problems-master/LongestPalindromicSubstringLength/LongestPalindromicSubstringLength.java
import java.util.*;
public class LongestPalindromicSubstringLength  
{ 
    static int longestPalindromicSubstringLength(String str) { 
// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        int n = str.length();
        boolean table[][] = new boolean[n][n]; 
        int maxLength = 1; 
        for (int i = 0; i < n; ++i) 
            table[i][i] = true; 
        int start = 0; 
        for (int i = 0; i < n - 1; ++i) { 
            if (str.charAt(i) == str.charAt(i + 1)) { 
                table[i][i + 1] = true; 
                start = i; 
                maxLength = 2; 
            } 
        } 
        for (int k = 3; k <= n; ++k) { 
            for (int i = 0; i < n - k + 1; ++i)  
            {  
                int j = i + k - 1; 
                if (table[i + 1][j - 1] && str.charAt(i) == str.charAt(j)) { 
                    table[i][j] = true; 
                    if (k > maxLength) { 
                        start = i; 
                        maxLength = k; 
                    } 
                } 
            } 
        } 
        return maxLength;
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    } 
    public static void main(String[] args) { 
        Scanner sc = new Scanner(System.in);
        int t= sc.nextInt();
        for(int tests=0;tests<t;tests++)
        {
            int n= sc.nextInt();
            String s= sc.next();
            System.out.println(longestPalindromicSubstringLength(s));
        }
    } 
}
